numba/generators.py

```diff
@@ -326,7 +316,14 @@ def lower_yield_suspend(self):
                                               0, state_index)
             ty = self.gentype.state_types[state_index]
             val = self.lower.loadvar(name)
+            # Load and DecRef previously stored value
+            oldval = self.context.unpack_value(self.builder, ty, state_slot)
+            if self.context.enable_nrt:
+                self.lower.decref(ty, oldval)
```

Comment:
If oldval and val are the same, decref before incref risks destroying the underlying data, no?